scaaml/capture/scope/lecroy/lecroy_waveform.py

```diff
@@ -0,0 +1,388 @@
+        self.d["nom_subarray_count"] = self.get_int16()
+        self.d["horizontal_interval"] = self.get_float()
+        self.d["horizontal_offset"] = self.get_double()
+        self.d["pixel_offset"] = self.get_double()
+        self.d["vertical_unit"] = self.get_string(48)
+        self.d["horizontal_unit"] = self.get_string(48)
+        self.d["horizontal_uncertainty"] = self.get_float()
+        self.d["trigger_time"] = self.get_timestamp()
+        self.d["acq_duration"] = self.get_float()
+
+        # Record type
+        self.d["record_type"] = {
+            0: "single_sweep",
+            1: "interleaved",
+            2: "histogram",
+            3: "graph",
+            4: "filter_coefficient",
+            5: "complex",
+            6: "extrema",
+            7: "sequence_obsolete",
+            8: "centered_RIS",
+            9: "peak_detect"
+        }.get(self.get_int16())
+
+        # Processing done
+        self.d["processing_done"] = {
+            0: "no_processing",
+            1: "fir_filter",
+            2: "interpolated",
+            3: "sparsed",
+            4: "autoscaled",
+            5: "no_result",
+            6: "rolling",
+            7: "cumulative"
+        }.get(self.get_int16())
+
+        self.get_int16()  # Reserved
+        self.d["ris_sweeps"] = self.get_int16()
+
+        # Timebase
+        self.d["timebase"] = {
+            0: "1 ps/div",
+            1: "2 ps/div",
+            2: "5 ps/div",
+            3: "10 ps/div",
+            4: "20 ps/div",
+            5: "50 ps/div",
+            6: "100 ps/div",
+            7: "200 ps/div",
+            8: "500 ps/div",
+            9: "1 ns/div",
+            10: "2 ns/div",
+            11: "5 ns/div",
+            12: "10 ns/div",
+            13: "20 ns/div",
+            14: "50 ns/div",
+            15: "100 ns/div",
+            16: "200 ns/div",
+            17: "500 ns/div",
+            18: "1 us/div",
+            19: "2 us/div",
+            20: "5 us/div",
+            21: "10 us/div",
+            22: "20 us/div",
+            23: "50 us/div",
+            24: "100 us/div",
+            25: "200 us/div",
+            26: "500 us/div",
+            27: "1 ms/div",
+            28: "2 ms/div",
+            29: "5 ms/div",
+            30: "10 ms/div",
+            31: "20 ms/div",
+            32: "50 ms/div",
+            33: "100 ms/div",
+            34: "200 ms/div",
+            35: "500 ms/div",
+            36: "1 s/div",
+            37: "2 s/div",
+            38: "5 s/div",
+            39: "10 s/div",
+            40: "20 s/div",
+            41: "50 s/div",
+            42: "100 s/div",
+            43: "200 s/div",
+            44: "500 s/div",
+            45: "1 ks/div",
+            46: "2 ks/div",
+            47: "5 ks/div",
+            100: "external"
+        }.get(self.get_int16())
+
+        # Vertical coupling
+        self.d["vertical_coupling"] = {
+            0: "DC_50_Ohms",
```

Comment:
Same here, for internal storage, let's use values that allows us to setup the scope in a straightforward way and do some pretty print when the context requires it

Reply:
I opted for storing A1M, D1M, D50, GND and showing those as well (quite human readable IMO). If the previous are better I can do pretty printing.